### PR TITLE
Add fraction.js, sicmutils.ratio for cross-platform ratio support [in progress]

### PR DESCRIPTION
**Pull request by sritchie**
It's time. I'll be able to get to this in the next couple of days...

EDIT: This could be a much better ratio literal for cljs than the mess I have now. https://github.com/mfikes/precise/blob/master/src/precise/tagged_literals.cljc#L17

### CODE REVIEW

File: src/sicmutils/ratio.cljc [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+     (defmethod op [Ratio ::v/integral] [a b] (f a b))
+     (defmethod op [::v/integral Ratio] [a b] (f a b)))
+
+   :cljs
+   (do
+     ;; The -equiv implementation handles equality with any number, so flip the
+     ;; arguments around and invoke equiv.
+     (defmethod v/eq [::v/number Fraction] [l r] (= r l))
+
+     (defn promote [x]
```
Comment (littleredcomputer) [inline src/sicmutils/ratio.cljc:166]: I can't figure out if it's a promotion or a demotion :)

File: src/sicmutils/ratio.cljc [hunk 2]
```diff
@@ -0,0 +1,223 @@
+;;
+;; Copyright © 2017 Colin Smith.
```
Comment (littleredcomputer) [inline src/sicmutils/ratio.cljc:2]: Maybe we should replace this with your name, or "the sicmutils authors" the way they do in golang

Comment (sritchie) [reply src/sicmutils/ratio.cljc:2]: I'm fine with this, or "sicmutils authors", either way!

**Comment by sritchie** [comment]
Closing in favor of #94 , #90 , #89 , #95 and #96 .

**Pull request closed without merging**